packages/itwinui-react/src/core/LabeledInput/LabeledInput.tsx

```diff
@@ -83,29 +99,51 @@ export const LabeledInput = React.forwardRef((props, ref) => {
   const icon = svgIcon ?? (status && StatusIconMap[status]());
 
   return (
-    <InputContainer
-      label={label}
-      disabled={disabled}
-      required={required}
-      status={status}
-      message={message}
-      icon={icon}
-      isLabelInline={displayStyle === 'inline'}
-      isIconInline={iconDisplayStyle === 'inline'}
-      className={className}
-      style={style}
-      inputId={id}
-    >
-      <Input
-        disabled={disabled}
-        className={inputClassName}
-        style={inputStyle}
-        required={required}
-        ref={ref}
-        id={id}
-        {...rest}
-      />
-    </InputContainer>
+    <InputGrid labelPlacement={displayStyle} {...wrapperProps}>
+      {label && (
+        <Label
+          as='label'
+          required={required}
+          disabled={disabled}
+          htmlFor={id}
+          {...labelProps}
+        >
+          {label}
+        </Label>
+      )}
+
+      <InputWithDecorations
+        status={status}
+        isDisabled={disabled}
+        {...inputWrapperProps}
+      >
+        <InputWithDecorations.Input
+          disabled={disabled}
+          required={required}
+          id={id}
+          ref={ref}
+          {...rest}
+        />
+        {icon && iconDisplayStyle === 'inline' && (
+          <Icon fill={!svgIcon ? status : undefined} padded {...iconProps}>
+            {icon}
+          </Icon>
+        )}
+      </InputWithDecorations>
+
+      {typeof message === 'string' || !!icon ? (
```

Comment:
i'm confused what's going on here - why is the `!!icon` needed?

Reply:
This is not needed in input. Removed.
Keeping it in LabeledSelect - we have unit test case there.

Reply:
what's the use case in LabeledSelect? and which unit test are you talking about? i can't find it.

Reply:
This use case: 
<img width="465" alt="image" src="https://github.com/iTwin/iTwinUI/assets/83585998/eaf5b9e8-de4e-4080-8534-dd67aed4f099">
This test: https://github.com/iTwin/iTwinUI/blob/5c14bc7946b643102ce8d789cf9502c643c14270/packages/itwinui-react/src/core/LabeledSelect/LabeledSelect.test.tsx#L91

Reply:
This doesn't look like valid usage though. I'd be ok with with removing it, to reduce complexity

Reply:
Removed, test updated to have message.